controllers/v1/userExtensionController.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * name : userExtension.js
+ * author : Rakesh Kumar
+ * created-date : 19-march-2020
+ * Description  User related information.
+ */
+
+
+/**
+ * dependencies
+ * 
+ */
+const userExtensionHelper = require(ROOT_PATH + "/module/userExtension/helper");
+
+
+
+/**
+    * UserExtension
+    * @class
+*/
+
+
+
+module.exports = class UserExtension extends Abstract {
+
+    constructor() {
+        super(userExtensionSchema);
+   
+    }
+
+    /**
+     * @apiDefine errorBody
+     * @apiError {String} status 4XX,5XX
+     * @apiError {String} message Error
+     */
+
+    /**
+     * @apiDefine successBody
+     *  @apiSuccess {String} status 200
+     * @apiSuccess {String} result Data
+     */
+
+
+    static get name() {
+        return "userExtension";
+    }
+
+
+
+      /**
+  * @api {get} /user-management-service/api/v1/userExtension/getProfile/{{userId}} Get user profile
+  * @apiVersion 1.0.0
+  * @apiName Get user profile
+  * @apiGroup User Extension
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /user-management-service/api/v1/userExtension/getProfile/e97b5582-471c-4649-8401-3cc4249359bb
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  * {
+      "_id": "5da6e08f436f9f3cd80b57b9",
+      "roles": [
+          {
+              "roleId": "5da580c746b88419104d8728",
+              "code": "OBS_DESIGNER"
+          },
+          {
+              "roleId": "5da580dc46b88419104d8737",
+              "code": "OBS_REVIEWERS"
+          }
+      ],
+      "updatedBy": "e97b5582-471c-4649-8401-3cc4249359bb",
+      "createdBy": "e97b5582-471c-4649-8401-3cc4249359bb",
+      "userId": "e97b5582-471c-4649-8401-3cc4249359bb",
+      "username": "a1",
+  * }
+  */
+
+  getProfile(req) {
+    return new Promise(async (resolve, reject) => {
+
+      try {
+
+        let queryObject = {
```

Comment:
@rakeshSgr , @deepakr123 - I would suggest pass just the user id from here, let query object and status filter be in the helper